### PR TITLE
add oasim variables to ModelData::defaultVariables and modifying pres var name 

### LINKED ISSUE #1076: add oasim variables to ModelData and modifying pres var name
## Description
In order to make coupled oasim work we need to change pressure variable name to mach with the fv3-jedi's naming and also we need to add oasim input variables to ModelData. for now just surface wind speed and surface pressure variables are read from gfs and the rest of the oasim input variables are coming from soca. This is going to change when we figure out how to derive all the atmospheric variables from gfs. this issue is need to be addressed for https://github.com/JCSDA-internal/skylab/issues/507

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support https://github.com/JCSDA-internal/skylab/issues/507
- needed by https://github.com/JCSDA-internal/skylab/issues/507

### PR DESCRIPTION
**Pull request by HamidehGMAO**
## Description

In order to make coupled oasim work we need to change pressure variable name to mach with the fv3-jedi's naming and also we need to add oasim input variables to ModelData. for now just surface wind speed and surface pressure variables are read from gfs and the rest of the oasim input variables are coming from soca. This is going to change when we figure out how to derive all the atmospheric variables from gfs. this needs to be merged with or after https://github.com/JCSDA-internal/ufo/pull/3470

build-group=https://github.com/JCSDA-internal/ufo/pull/3470

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #1076 1076

## Testing

How were these changes tested? with running coupled-gfs-mom6-hofx.yaml in corresponding skylab branch
What compilers / HPCs was it tested with? discover
Are the changes covered by ctests? the pressure variable name change is covered by 49: test_soca_hofx_oasim_3d, and the ModelData changes are covered by oupled-gfs-mom6-hofx.yaml in skylab

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names for the automated TravisCI tests to pass

Do PRs in upstream repositories need to be merged first?

- waiting on https://github.com/JCSDA-internal/ufo/pull/3470

### CODE REVIEW

**Comment by travissluka** [comment]
@HamidehGMAO Please add the proper `build-group=`  flags in the description for all related PRs, (see https://github.com/JCSDA-internal/soca/pull/1077/checks?check_run_id=30608919123) otherwise the CI tests will fail

**Comment by HamidehGMAO** [comment]
@travissluka ah! I see! I will do so.

**Review by svahl991 — COMMENTED** [review]
Looks fine, except the coding norms ctest is failing.

File: src/soca/ModelData/ModelData.h [hunk 1]
```diff
@@ -25,7 +25,16 @@ class ModelData : public util::Printable {
  public:
   static const std::string classname() {return "soca::ModelData";}
   static const oops::Variables defaultVariables() {
-    return oops::Variables(std::vector<std::string>({"surface_temperature_where_sea"}));
+    return oops::Variables(std::vector<std::string>({"surface_temperature_where_sea",
```
Comment (travissluka) [inline src/soca/ModelData/ModelData.h:28]: similar to how the `test/Data/fields_metadata.yml` file is split up by variable type (bgc, atmosphere, etc), could you do the same here? Mainly, keep the atmospheric variables separate, with a comment before them stating that they're atmosphere, and should eventually be deleted

**Review by travissluka — COMMENTED** [review]
one comment. Also, when you have time, it would be good to issue a PR to enable OASIM in the coupling repo, otherwise these changes will not be tested in the CI integration tests..

**Review by travissluka — APPROVED** [review]

**Review by shlyaeva — APPROVED** [review]

**Comment by travissluka** [comment]
@eap our integration tests, a substantial percentage of the time, randomly fail due to repos such as `femps` not being able to clone from github (https://jedi-ci-build-public-logs.s3.us-east-2.amazonaws.com/soca-1077-e74e144-intel-UDNW3NVI.html). Is this a known issue? It's been happening for several months now.

**Comment by shlyaeva** [comment]
> @eap our integration tests, a substantial percentage of the time, randomly fail due to repos such as femps not being able to clone from github (https://jedi-ci-build-public-logs.s3.us-east-2.amazonaws.com/soca-1077-e74e144-intel-UDNW3NVI.html). Is this a known issue? It's been happening for several months now.

@eap I've seen it multiple times in other repositories as well.

**Pull request merged**